### PR TITLE
[AMORO-2009]: Add aws/s3/hadoop-aws and iceberg-aws as default dependenies.

### LINKED ISSUE #2009: [Improvement]: Support `s3://` and `s3a://` protocols in the default distribution.
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Include the aws-bundle/hadoop-aws/hadoop-client package in the default distribution to support mainstream storage protocols.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?
 
Close #2009.

## Brief change log
 
- Add `iceberg-aws` as default dependenies for all module
- Add `hadoop-aws`, `awssdk` to AMS module depndenies.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? ( no)

### CODE REVIEW

File: ams/optimizer/flink-optimizer/pom.xml [hunk 1]
```diff
@@ -177,6 +177,11 @@
             </exclusions>
         </dependency>
 
+        <dependency>
```
Comment (zhoujinsong) [inline ams/optimizer/flink-optimizer/pom.xml:180]: Does the `flink-optimizer` module still need to declear the dependency as it had dependent on `amoro-core`?

File: ams/server/pom.xml [hunk 2]
```diff
@@ -219,6 +219,42 @@
             <artifactId>kyuubi-hive-jdbc-shaded</artifactId>
         </dependency>
 
+        <!-- aws dependencies -->
+        <dependency>
```
Comment (zhoujinsong) [inline ams/server/pom.xml:223]: I am not sure why specific AWS dependencies are only added to the `ams-server` module. Are these dependencies not needed for other modules, such as the `StandaloneOptimizer`?

File: spark/v3.1/spark-runtime/pom.xml [hunk 3]
```diff
@@ -76,6 +76,7 @@
                                     <include>org.apache.iceberg:iceberg-data</include>
                                     <include>org.apache.iceberg:iceberg-orc</include>
                                     <include>org.apache.iceberg:iceberg-parquet</include>
+                                    <include>org.apache.iceberg:iceberg-aws</include>
```
Comment (zhoujinsong) [inline spark/v3.1/spark-runtime/pom.xml:79]: Should we add this to `flink-runtime` module too?

File: ams/server/pom.xml [hunk 4]
```diff
@@ -219,6 +219,42 @@
...
+
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>url-connection-client</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-aws</artifactId>
+        </dependency>
```
Comment (XBaith) [inline ams/server/pom.xml:256]: What are the scenarios where we need to use hadoop-aws `dependencies`

File: core/pom.xml [hunk 5]
```diff
@@ -189,50 +194,4 @@
...
-                    <artifactId>sts</artifactId>
-                </dependency>
-
-                <dependency>
-                    <groupId>software.amazon.awssdk</groupId>
-                    <artifactId>url-connection-client</artifactId>
-                </dependency>
-            </dependencies>
-        </profile>
-    </profiles>
```
Comment (XBaith) [inline core/pom.xml:236]: One thing I'm not so sure about is: `amoro-core` is used in multiple modules, will moving to the `aws-server` module affect the service?

File: ams/server/pom.xml (hunk 2 above)
Comment (baiyangtx) [reply ams/server/pom.xml:223]: StandaloneOptimizer  is not a fatjar and event not a dependent plugin. It was put to ./lib and using the AMS classpath.

aws dependencies should be controlled by engines.

File: spark/v3.1/spark-runtime/pom.xml (hunk 3 above)
Comment (baiyangtx) [reply spark/v3.1/spark-runtime/pom.xml:79]: Not, the flink modules using  `org.apache.iceberg:* ` as filter

File: ams/server/pom.xml (hunk 4 above)
Comment (baiyangtx) [reply ams/server/pom.xml:256]: to support `s3a://`  file system with hive

File: core/pom.xml (hunk 5 above)
Comment (baiyangtx) [reply core/pom.xml:236]: aws sdk shoud be controlled by compute engines.  Amoro connectors should not shade them.

File: ams/optimizer/flink-optimizer/pom.xml (hunk 1 above)
Comment (baiyangtx) [reply ams/optimizer/flink-optimizer/pom.xml:180]: I will fix this.

**Comment by baiyangtx** [comment]
Works in locally and minio standalone cluster. 

图片

图片

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Review by wangtaohz — COMMENTED** [review]
LGTM. :+1:

We can merge it ASAP, which would be helpful for my local testing.
